Add support for Weatherbit in the Weather Module

AndyPoms opened the pull request:
This is my first ever pull request, please let me know if I messed anything up.

Adds support for Weatherbit.io
Based on existing classes for Dark Sky and Weather.gov

***API Key required***

Config Options:

Option | Description
---|---
`apiBase` | The Weather base URL.   Possible value: `https://api.weatherbit.io/v2.0`  This value is **REQUIRED**
`weatherEndpoint` | The Weatherbit API endPoint.  Possible values: `/current`, `/forecast/daily`  This value is **REQUIRED**
`apiKey` | The Weatherbit API key which can be obtained by creating an WeatherBit account at https://www.weatherbit.io  This value is **REQUIRED**
`lat` | The geo coordinate latitude.  This value is **REQUIRED**
`lon` | The geo coordinate longitude.  This value is **REQUIRED**

rejas commented on modules/default/weather/providers/weatherbit.js:
```diff
@@ -0,0 +1,181 @@
+	// Implement WeatherDay generator.
+	generateWeatherDayFromCurrentWeather(currentWeatherData) {
+		//Calculate TZ Offset and invert to convert Sunrise/Sunset times to Local
+		var d = new Date();
```
can be a const ?

rejas commented on modules/default/weather/providers/weatherbit.js:
```diff
@@ -0,0 +1,181 @@
+	generateWeatherDayFromCurrentWeather(currentWeatherData) {
+		//Calculate TZ Offset and invert to convert Sunrise/Sunset times to Local
+		var d = new Date();
+		var tzOffset = d.getTimezoneOffset();
```
should be a let

rejas commented:
As the bot says: Please include an entry for your PR in the Changelog. 
Otherwise I found only two small nitbits looking at the code, dont have the time unfortanetly to run it and see if it works :-)

rejas commented:
Also, you should run "npm run lint:prettier" to make sure your code adheres to the styleguides (and passes the travis tests)

AndyPoms commented:
Like I said this is my first ever pull request...  How do I include a CHANGELOG.md file?  What should it include? 

Do I just take the one from my fork, add a line that says "Add support for Weatherbit" & attach?  Do I create a new file named CHANGELOG.md with only a line that say "Adds support for Weatherbit"?

How do I run `npm run lint:prettier`?  Is it on the website, the Desktop app, or GitBash, GitCMD, or GitGUI?

@rejas I'll review your const/let comments when I get a chance (likely towards the end of the week).

rejas commented:
> Like I said this is my first ever pull request... How do I include a CHANGELOG.md file? What should it include?
> 
> Do I just take the one from my fork, add a line that says "Add support for Weatherbit" & attach? Do I create a new file named CHANGELOG.md with only a line that say "Adds support for Weatherbit"?

You should just add a new line to the already existing CHANGELOG.md file under the "Added" headline.

> How do I run `npm run lint:prettier`? Is it on the website, the Desktop app, or GitBash, GitCMD, or GitGUI? Or is it on the machine I'm hosting my MM with?
You run that command on the command line on the machine you are developing and submitting your PR. Just go into the main directory of the MagicMirror repository.

> @rejas I'll review your const/let comments when I get a chance (likely towards the end of the week).
:+1:

sdetweil commented:
>>How do I run npm run lint:prettier

it's a commandline
all git and npm commands are from the MagicMirror folder

AndyPoms commented:
- Updated to const & let per rejas suggestions.
- Added CHANGELOG.md (I think I may have screwed this one up based on the Conflict listed above - advice?)
- Ran `npm run lint:prettier`

Let me know if there is anything else

rejas commented:
Just fix that one merge-conflict and the PR seems good to go!

AndyPoms commented:
> 
> 
> Just fix that one merge-conflict and the PR seems good to go!

I think I did it correctly & kept all changes (just removed the 3 conflict markers so that all info remained.

rejas commented:
Please run "npm run lint:prettier" to fix the style issues in the CHANGELOG. After those changes are pushed, this PR should be fine.

AndyPoms commented:
> 
> 
> Please run "npm run lint:prettier" to fix the style issues in the CHANGELOG. After those changes are pushed, this PR should be fine.

Done....  Weird because my original CHANGELOG was OK, it was only after I resolved a conflict (by keeping both sets of changes) that this came up....
